From 4c1070add5dd6453f3c0749d7e1958f81c5c2671 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sun, 15 Dec 2024 11:21:54 +0800
Subject: [PATCH 251/261] AOSCOS: audit: remove duplicate functions for MIPS

audit_classify_arch() and audit_classify_syscall() are already defined
in arch/mips/kernel/audit-native.c, adding preprocessor directives to
remove them from lib/audit.c on MIPS.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 lib/audit.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/audit.c b/lib/audit.c
index 738bda22dd39..5f68998fdaf6 100644
--- a/lib/audit.c
+++ b/lib/audit.c
@@ -29,6 +29,7 @@ static unsigned signal_class[] = {
 ~0U
 };
 
+#if !defined(CONFIG_MIPS)
 int audit_classify_arch(int arch)
 {
 	if (audit_is_compat(arch))
@@ -68,6 +69,7 @@ int audit_classify_syscall(int abi, unsigned syscall)
 		return AUDITSC_NATIVE;
 	}
 }
+#endif
 
 static int __init audit_classes_init(void)
 {
-- 
2.48.1

